src/main/duke/data/task/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package duke.data.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * This class abstracts a deadline task.
+ */
+public class Deadline extends Task {
+    protected final LocalDateTime by;
```

Comment:
Final variables/constants should be in full capital letters according to the Java Coding Standard.